.github/actions/setup-lxd-cluster/lxd-cluster.sh

```diff
@@ -0,0 +1,372 @@
+VERSION_LXD="${VERSION_LXD:-latest/edge}"
+
+# Other.
+INSTANCE="${CLUSTER_NAME}"
+LEADER="${CLUSTER_NAME}-1"
+STORAGE_POOL="${CLUSTER_NAME}-pool"
+STORAGE_DRIVER="${STORAGE_DRIVER:-dir}"
+NETWORK_NAME="${CLUSTER_NAME}br0"
+
+# Source bin/helpers from canonical/lxd-ci repository.
+# shellcheck source=/dev/null
+source <(
+  curl -fsSL https://raw.githubusercontent.com/canonical/lxd-ci/refs/heads/main/bin/helpers \
+  || { echo "Error: Failed to source bin/helpers from canonical/lxd-ci" >&2; exit 1; }
+)
+
+#================================================
+# Utils
+#================================================
+
+# instanceIPv4 returns the IPv4 address of the instance with the given name.
+instanceIPv4() {
+    local instance="$1"
+    local ipv4
+
+    # Try to get any IPv4 address from preconfigured bridge br0.
+    ipv4=$(lxc ls "${instance}" -f csv -c 4 | grep "(br0)" | awk '{print $1}' || true)
+    if [ "${ipv4}" != "" ]; then
+        echo "${ipv4}"
+        return 0
+    fi
+
+
+    # Try for enp5s0 (VM) and eth0 (container) interfaces.
+    for ifName in enp5s0 eth0; do
+        ipv4=$(lxc ls "${instance}" -f csv -c 4 | grep -oP "(\d{1,3}\.){3}\d{1,3}(?= \(${ifName}\))" || true)
+        if [ "${ipv4}" != "" ]; then
+            echo "${ipv4}"
+            return 0
+        fi
+    done
+
+    echo "Error: Failed to retrieve IPv4 address for instance ${instance}" >&2
+    return 1
+}
+
+#========================
+# Cluster setup
+#========================
+
+# deploy deploys instances required for a LXD cluster.
+deploy() {
+    local instance
+
+    if ! lxc network show "${NETWORK_NAME}" &>/dev/null; then
+        echo "Creating network ${NETWORK_NAME} ..."
+        lxc network create "${NETWORK_NAME}" ipv4.address=172.16.20.1/24 ipv4.nat=true
+    fi
+
+    # Create storage pool.
+    echo "Creating storage pool ${STORAGE_POOL} ..."
+    if ! lxc storage show "${STORAGE_POOL}" &>/dev/null; then
+        if [ "${STORAGE_DRIVER}" = "zfs" ]; then
+            lxc storage create "${STORAGE_POOL}" zfs volume.zfs.delegate=true
+
+            # XXX: Ensure that the ZFS device is accessible within the nested container.
+            #      Otherwise, the LXD storage pool creation will fail.
+            local zfsPerm
+            zfsPerm=$(stat -c '%a' /dev/zfs)
+
+            if [ $((zfsPerm & 7)) -eq 0 ]; then
+                # It's udev's job to fix the perms but the rule for it ships in
+                # the zfsutils-linux package which might not be installed.
+                sudo chmod 0660 /dev/zfs
+            fi
+        else
+            lxc storage create "${STORAGE_POOL}" "${STORAGE_DRIVER}"
+        fi
+    fi
+
+    # Create container profile capable of running VMs.
+    if ! lxc profile show container-kvm &>/dev/null; then
+        lxc profile create container-kvm << EOF
+description: Makes containers capable of running VMs
+config:
+  linux.kernel_modules: kvm,vhost_net,vhost_vsock
+  security.devlxd.images: "true"
```

Comment:
That's already provided at `lxc init` time due to it being needed on VMs too. As such, I'd drop it from the profile.